.travis.yml

```diff
@@ -5,41 +5,49 @@ notifications:
-  - commit_id=$(sudo docker ps -l | tail -n1 | awk '{print $1}')
-  # make the build directory
-  - sudo docker run cnerg/dagmc-ci-ubuntu-16.04 /bin/bash -c "mkdir -p /root/build/$COMPILER/DAGMC-moab-$MOAB_VERSION"
-  - commit_id=$(sudo docker ps -l | tail -n1 | awk '{print $1}')
-  - sudo docker commit $commit_id cnerg/dagmc-ci-ubuntu-16.04
-  # copy the repo in the docker instance
-  - sudo docker cp ../DAGMC $commit_id:/root/build/$COMPILER/DAGMC-moab-$MOAB_VERSION
-  - commit_id=$(sudo docker ps -l | tail -n1 | awk '{print $1}')
-  - sudo docker commit $commit_id cnerg/dagmc-ci-ubuntu-16.04
+  - docker_image=ljacobson64/dagmc-ci-ubuntu-${UBUNTU_VERSION}
+  - build_dir=/root/build/${COMPILER}/DAGMC-moab-${MOAB_VERSION}-hdf5-${HDF5_VERSION}
+  # Get the docker image
+  - docker pull ${docker_image}
+  # Make the build directory
+  - docker run ${docker_image} /bin/bash -c "mkdir -p ${build_dir}"
+  - commit_id=$(docker ps -l | tail -n1 | awk '{print $1}')
+  - docker commit ${commit_id} ${docker_image}
+  # Copy the repo in the docker instance
+  - docker cp ../DAGMC ${commit_id}:${build_dir}
+  - commit_id=$(docker ps -l | tail -n1 | awk '{print $1}')
+  - docker commit ${commit_id} ${docker_image}
 install:
-  - sudo docker run cnerg/dagmc-ci-ubuntu-16.04 /bin/bash -c "cd /root/build/$COMPILER/DAGMC-moab-$MOAB_VERSION/DAGMC;
-                                                                    bash docker/travis.install.sh $ASTYLE_ONLY $COMPILER $MOAB_VERSION $BUILD_GEANT4"
-  - commit_id=$(sudo docker ps -l | tail -n1 | awk '{print $1}')
-  - sudo docker commit $commit_id cnerg/dagmc-ci-ubuntu-16.04
+  # Build DAGMC
+  - docker run ${docker_image} /bin/bash -c "if [ \"${HOUSEKEEPING_ONLY}\" == \"false\" ]; then
+                                               bash ${build_dir}/DAGMC/docker/travis.install.sh \"${COMPILER}\" \"${HDF5_VERSION}\" \"${MOAB_VERSION}\";
+                                             fi"
+  - commit_id=$(docker ps -l | tail -n1 | awk '{print $1}')
+  - docker commit ${commit_id} ${docker_image}
 script:
-  - sudo docker run cnerg/dagmc-ci-ubuntu-16.04 /bin/bash -c "cd /root/build/$COMPILER/DAGMC-moab-$MOAB_VERSION/DAGMC;
-                                                                    bash docker/travis.script.sh $ASTYLE_ONLY $COMPILER $MOAB_VERSION $BUILD_GEANT4"
+  # Run unit tests
+  - docker run ${docker_image} /bin/bash -c "if [ \"${HOUSEKEEPING_ONLY}\" == \"true\" ]; then
+                                               bash ${build_dir}/DAGMC/docker/travis.housekeeping.sh \"${COMPILER}\" \"${HDF5_VERSION}\" \"${MOAB_VERSION}\" \"${TRAVIS_REPO_SLUG}\" \"${TRAVIS_PULL_REQUEST}\";
+                                             else
+                                               bash ${build_dir}/DAGMC/docker/travis.tests.sh        \"${COMPILER}\" \"${HDF5_VERSION}\" \"${MOAB_VERSION}\" \"${TRAVIS_REPO_SLUG}\" \"${TRAVIS_PULL_REQUEST}\";
+                                             fi"
```

Comment:
It might be more human readable if you split this into 2 travis command, and do the if statement in `travis` instead of in the docker container, something like:
```
# Housekeeping
- if ${HOUSEKEEPING_ONLY} then 
 docker run ${docker_image} /bin/bash -c " travis.housekeeping.sh ...."
 fi
# Run unit tests
-  if ! ${HOUSEKEEPING_ONLY} then 
 docker run ${docker_image} /bin/bash -c " /travis.tests.sh ...."
 fi

Reply:
Good idea. I will change this. For some reason it didn't occur to me that you could have an `if` statement in `travis` directly.

Reply:
This is done.